File: tests/csharp/csharp_integer_arithmetic/division_on_double_variables_does_not_truncate.cs
// vybe-test: csharp/csharp_integer_arithmetic/division_on_double_variables_does_not_truncate
// origin: languages/csharp/tests/csharp/test_csharp_integer_arithmetic.rs

using static __Harness;

double a = 9, b = 4;
__P((a / b).ToString());
__Check("2.25");

public static class __Harness {
    public static string __buf = "";
    public static void __P(string s) { __buf = __buf + s + "\n"; }
    public static void __Pr(string s) { __buf = __buf + s; }
    public static void __Check(string want) {
        if (__buf != want && __buf != want + "\n") {
            Console.WriteLine("FAIL: want [" + want + "] got [" + __buf + "]");
            throw new Exception("assertion failed");
        }
    }
}

File: tests/c/operators_arithmetic/division_on_double_variables_does_not_truncate.c
// vybe-test: c/operators_arithmetic/division_on_double_variables_does_not_truncate
// origin: languages/c/tests/c/test_operators_arithmetic.rs
#include <stdio.h>
#include <string.h>
#include <assert.h>
int main() {
const char *__w[] = {"3.5\n"};
int __n = 1, __i = 0;
double a = 7.0, b = 2.0;
{ char __t[512]; snprintf(__t, sizeof(__t), "%.1f\n", a / b);
  if (__i >= __n || strcmp(__t, __w[__i]) != 0) { printf("FAIL at line %d: got [%s]\n", __i, __t); assert(0); } __i++; }
if (__i != __n) { printf("FAIL: %d line(s), wanted %d\n", __i, __n); assert(0); }
return 0;
}

